homeassistant/components/schlage/switch.py

```diff
@@ -0,0 +1,124 @@
+from .entity import SchlageEntity
+
+
+@dataclass
+class SchlageSwitchEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    # NOTE: This has to be a mixin because these are required keys.
+    # SwitchEntityDescription has attributes with default values,
+    # which means we can't inherit from it because you haven't have
+    # non-default arguments follow default arguments in an initializer.
+
+    on_fn: Callable[[Lock], None]
+    off_fn: Callable[[Lock], None]
+    value_attr: str
+
+
+@dataclass
+class SchlageSwitchEntityDescription(
+    SwitchEntityDescription, SchlageSwitchEntityDescriptionMixin
+):
+    """Entity description for a Schlage switch."""
+
+
+SWITCHES: tuple[SchlageSwitchEntityDescription, ...] = (
+    SchlageSwitchEntityDescription(
+        key="beeper",
+        translation_key="beeper",
+        device_class=SwitchDeviceClass.SWITCH,
+        entity_category=EntityCategory.CONFIG,
+        on_fn=lambda lock: lock.set_beeper(True),
+        off_fn=lambda lock: lock.set_beeper(False),
+        value_attr="beeper_enabled",
+    ),
+    SchlageSwitchEntityDescription(
+        key="lock_and_leve",
+        translation_key="lock_and_leave",
+        device_class=SwitchDeviceClass.SWITCH,
+        entity_category=EntityCategory.CONFIG,
+        on_fn=lambda lock: lock.set_lock_and_leave(True),
+        off_fn=lambda lock: lock.set_lock_and_leave(False),
+        value_attr="lock_and_leave_enabled",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+    for device_id in coordinator.data.locks:
+        for description in SWITCHES:
+            entities.append(
+                SchlageSwitch(
+                    coordinator=coordinator,
+                    description=description,
+                    device_id=device_id,
+                )
+            )
+    async_add_entities(entities)
+
+
+class SchlageSwitch(SchlageEntity, SwitchEntity):
+    """Schlage switch entity."""
+
+    entity_description: SchlageSwitchEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SchlageDataUpdateCoordinator,
+        description: SchlageSwitchEntityDescription,
+        device_id: str,
+    ) -> None:
+        """Initialize a SchlageSwitch."""
+        super().__init__(coordinator=coordinator, device_id=device_id)
+        self.entity_description = description
+        self._attr_unique_id = f"{device_id}_{self.entity_description.key}"
+        self._attr_is_on = getattr(self._lock, self.entity_description.value_attr)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self._attr_is_on = getattr(self._lock, self.entity_description.value_attr)
+        return super()._handle_coordinator_update()
```

Comment:
Have you thought about changing this into a `is_on` property?

Reply:
Not a bad idea. Done.